x/gamm/pool-models/stableswap/amm.go

```diff
@@ -191,7 +191,8 @@ func spotPrice(baseReserve, quoteReserve sdk.Dec) sdk.Dec {
 
 // returns outAmt as a decimal
 func (pa *Pool) calcOutAmtGivenIn(tokenIn sdk.Coin, tokenOutDenom string, swapFee sdk.Dec) (sdk.Dec, error) {
-	reserves, err := pa.getPoolAmts(tokenIn.Denom, tokenOutDenom)
+	reserves, err := pa.getScaledPoolAmts(tokenIn.Denom, tokenOutDenom)
```

Comment:
We need to be 'descaling' all the pool amounts for the return values as well

Reply:
Oh good catch, thx!

Reply:
And again, I'm surprised that current testcases didn't catch this! 😅